Formula/docksal.rb

```diff
@@ -0,0 +1,25 @@
+class Docksal < Formula
+  desc "Docker powered web development environments for macOS, Windows & Linux"
+  homepage "https://docksal.io"
+  url "https://github.com/docksal/docksal/archive/v1.12.3.tar.gz"
+  sha256 "252a8ce0194fbcacb0a2b7dd6d578843f4ada2cd31c1859350a53791827932a0"
+
+  def install
+    mkdir_p "~/.docksal"
+    touch "~/.docksal/docksal.env"
+    system "echo", "'DOCKER_NATIVE=1'", ">>", "~/.docksal/docksal.env"
+    bin.install "bin/fin"
+  end
+
+  def caveats
+    s = "VirtualBox can be installed via `brew cask install virtualbox`\n"
+    s += "Docker can be installed via `brew cask install docker`\n"
+    s += "For Sandbox Server support run: `echo 'CI=1' >> ~/.docksal/docksal.env`\n"
+    s += "For Katacoda mode support run: `echo 'KATACODA=1' >> ~/.docksal/docksal.env`\n"
+    s
+  end
+
+  test do
+    system "#{bin}/fin", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@fxcoudert Even after reading the cookbook, I wasn't even thinking about testing failure. Taking your last point into mind I can definitely test for the failure of a docker instance failing to initialize.

Thanks again for taking the time to review this PR.